### ISSUE TITLE
Sidebar UI, W_NO_PADDING flag has issues

### ISSUE THREAD

**Issue by vetall812** [CleverRaven/Cataclysm-DDA#58598]
### Describe the bug

W_NO_PADDING flag breaks "arrange": "rows" in child widgets

### Steps to reproduce

Use attached sidebar.

### Expected behavior

Flag should only affect "arrange": "columns" widgets

### Screenshots

![image](https://user-images.githubusercontent.com/6676374/174816597-6263946a-2284-4fff-accf-7ee0f96d2b0d.png)

![image](https://user-images.githubusercontent.com/6676374/174817310-5248b6e1-1b87-4cda-a602-b2117deeec9f.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1766 (21H2)
- Game Version: 80cf483 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

[!_sidebar.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/8949537/_sidebar.zip)

related to the PR #58543

**Comment by dseguin** [comment]
This looks like a one-line fix:

```diff
diff --git a/src/widget.cpp b/src/widget.cpp
index eb09a3c488..5211250e98 100644
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -1471,7 +1471,7 @@ std::string widget::layout( const avatar &ava, unsigned int max_width, int label
                 const std::string txt = cur_child.layout( ava, skip_pad_this ? 0 : cur_width,
                                         label_width, skip_pad_this );
                 // Store the resulting text for this column
-                cols.emplace_back( foldstring( txt, skip_pad_this ? 0 : cur_width ) );
+                cols.emplace_back( string_split( txt, '\n' ) );
                 widths.emplace_back( cur_width );
             }
             int h_max = 0;
```

Seems to do the job:

![Screenshot from 2022-06-22 20-21-40](https://user-images.githubusercontent.com/12537966/175181847-fd0f2a5b-0251-44c2-94d5-e34d3df97a86.png)

**Comment by Inglonias** [comment]
It looks like I'm going to need this fix to resolve the regressions from #58113, because as I suspected, layouts are a pain in the ass.
